From bb2fb4b67d34316aac3be4bdb8f1e98123c86cf2 Mon Sep 17 00:00:00 2001
From: Tyler Smith <tyler.smith@windriver.com>
Date: Mon, 10 Sep 2018 19:11:33 +0000
Subject: [PATCH 06/18] Temporary workaround to deploying horizon via apache in
 CentOS vs Ubuntu

Changing launch command and modifying default CentOS config to
allow horizon to bind to port 80
---
 horizon/templates/bin/_horizon.sh.tpl | 25 ++++++++++++++++++-------
 horizon/templates/deployment.yaml     |  2 +-
 horizon/values.yaml                   |  6 ++++++
 3 files changed, 25 insertions(+), 8 deletions(-)

diff --git a/horizon/templates/bin/_horizon.sh.tpl b/horizon/templates/bin/_horizon.sh.tpl
index 8ff4a3c..f7fa029 100644
--- a/horizon/templates/bin/_horizon.sh.tpl
+++ b/horizon/templates/bin/_horizon.sh.tpl
@@ -29,12 +29,23 @@ function start () {
 
   a2enmod rewrite
 
-  if [ -f /etc/apache2/envvars ]; then
-     # Loading Apache2 ENV variables
-     source /etc/apache2/envvars
+  . /etc/os-release
+  if [ "x$ID" == "xubuntu" ]; then
+
+    if [ -f /etc/apache2/envvars ]; then
+       # Loading Apache2 ENV variables
+       source /etc/apache2/envvars
+    fi
+    rm -rf /var/run/apache2/*
+    APACHE_DIR="apache2"
+    APACHE_BINARY='apache2'
+
+  else
+
+    sed -i "s/Listen 80/#Listen 80/" /etc/httpd/conf/httpd.conf
+    APACHE_BINARY='httpd'
+
   fi
-  rm -rf /var/run/apache2/*
-  APACHE_DIR="apache2"
 
   # If the image has support for it, compile the translations
   if type -p gettext >/dev/null 2>/dev/null; then
@@ -46,11 +57,11 @@ function start () {
   /tmp/manage.py compress --force
   rm -rf /tmp/_tmp_.secret_key_store.lock /tmp/.secret_key_store
 
-  exec apache2 -DFOREGROUND
+  exec $APACHE_BINARY -DFOREGROUND
 }
 
 function stop () {
-  apachectl -k graceful-stop
+    kill -TERM 1
 }
 
 $COMMAND
diff --git a/horizon/templates/deployment.yaml b/horizon/templates/deployment.yaml
index fe6b8da..8de821f 100644
--- a/horizon/templates/deployment.yaml
+++ b/horizon/templates/deployment.yaml
@@ -90,7 +90,7 @@ spec:
               subPath: manage.py
               readOnly: true
             - name: horizon-etc
-              mountPath: /etc/apache2/sites-enabled/000-default.conf
+              mountPath: /etc{{ .Values.deployment.apache.path }}{{ .Values.deployment.apache.conf_path }}/000-default.conf
               subPath: horizon.conf
               readOnly: true
             - name: horizon-bin
diff --git a/horizon/values.yaml b/horizon/values.yaml
index 65d93b4..6f2aefc 100644
--- a/horizon/values.yaml
+++ b/horizon/values.yaml
@@ -17,6 +17,12 @@
 # Declare name/value pairs to be passed into your templates.
 # name: value
 
+deployment:
+  apache:
+    path: /apache2
+    conf_path: /conf-enabled
+    modules_path: /mods-available
+
 images:
   tags:
     db_init: docker.io/openstackhelm/heat:ocata
-- 
1.8.3.1

